### PR TITLE
Adjust Reloadly

### PR DESCRIPTION
**Pull request by calufa**
TODO:
- [x] Update staging.yaml with latests docker images
- [x] Move auth out of backoff logic
- [x] Create new docker image for `dinersclub`

### CODE REVIEW

File: dinersclub/fake.go [hunk 1]
```diff
@@ -18,11 +19,16 @@ func NewFakeProvider() (Provider, error) {
 func (p *FakeProvider) GetUserFromPaymentEvent(event *PaymentEvent) (*User, error) {
 	if event.Pageid == "page" || event.Pageid == "935593143497601" {
 		return &User{Id:"test-id"}, nil
+	} else if event.Pageid == "test-auth-page" {
```
Comment (nandanrao) [inline dinersclub/fake.go:22]: This feels circular to me. I'm too tired to figure out what actually needs to be done, but I feel like we've added a lot of logic for a test that mostly tests the logic we added. 

Fake is essentially a mock, so maybe it's ok, but this feels like what we want is a mock.

Comment (calufa) [reply dinersclub/fake.go]: I see what you mean. Just pushed a new commit, https://github.com/vlab-research/fly/pull/42/commits/4225846c02da510d460eceaf709f956f95e39bfa, let me know what do you think.

**Comment by nandanrao** [comment]
A) This is a much nicer solution!!! I like it. I can't even remember what the old one was, this looks so natural.

B) Actually had a bug today in production where it was silently failing because it was backoff-retrying on an auth error and then that was getting swallowed because I had the timeouts for the provider set longer than the kafka poll interval! Wish I woulda reviewed this earlier :D

**Comment by nandanrao** [comment]
Let's merge it!

**Pull request merged**